joomla/joomla-cms
# Allow users to share sessions between apps

### Summary of Changes

Joomla's default behavior when generating session names is to use the class name for the running application class, however it also supports using a configuration value for a "static" session name, and when this value is set a user can set their Joomla install to share sessions between the site and admin apps (or really any app class based on `JApplicationCms)`.  Let's expose this via the global configuration.
### Testing Instructions

With the patch applied, you'll see a new "Shared Sessions" config value under the session settings in the global configuration, defaulted to No (so off).  When you set this to on, you will be logged out of the admin (as explained in the tooltip) and will need to log in again.  Once logged in again, if you visit the frontend, you'll be authenticated.  The same goes if you log into the frontend then try accessing the admin.

Unfortunately the success message is lost because it is enqueued to the active session that saves the configuration and the redirected request will start a new session under the new name.  There isn't much that can be done here unfortunately without getting to a state where we pollute user sessions with data from another session.
### Documentation Changes Required

If someone had already enabled this by setting a `public $session_name = 'whatever';` var in their `configuration.php` file, they will need to add `public $shared_session = '1';` to it as well.  The latter var is needed to track the state of this option correctly and if it's not set you'll lose your shared session configuration.
